files/inputs/test36.json

```diff
@@ -0,0 +1,38 @@
+{
+"fooList": [
+	[
+	    [ 0.2,0.01,0.02,0.04],
+		[ 0.2,0.01,0.02],
+		[ 50.1,30.2],
+		[ 50.1,30.2,0.01,0.02,0.04],
+		[ 0.2,0.01,0.02],
+		[ 0.2,0.01,0.02,0.04]
+	],
+  ],
+"fooList3x": [
+	[
+	    [ 0.2,0.01,0.02,0.04],
+		[ 0.2,0.01,0.02],
+		[ 50.1,30.2],
+		[ 50.1,30.2,0.01,0.02,0.04],
+		[ 0.2,0.01,0.02],
+		[ 0.2,0.01,0.02,0.04]
```

Comment:
@porteri I have not looked into this in detail yet, however, my impression was that JSON-Fortran's extant matrix capabilities do not support sparse/ragged edge matrices. Is this a requirement for you? It will likely be easier to implement M by N matrices than ragged edge/sparse matrices in the JSON read & write routines.

Reply:
Yea unfortunately it is a requirement. That's why I added in the optional `matrix_column_size` flag. It works fine reading b/c I allocate it to the largest size of N (currently hardwired to 1000 and then reduced but will fix this in future)